lib/possible_bipartition.rb

```diff
@@ -1,4 +1,45 @@
 
 def possible_bipartition(dislikes)
-  raise NotImplementedError, "possible_bipartition isn't implemented yet"
+  return true if dislikes.empty?
+  graph_queue = []
+  searched = []
+  
+  color_array = []
+  dislikes.length.times do
+    color_array << 0
+  end
+  
+  dislikes.each_with_index { |enemies, dog|
+    graph_queue << dog
+  }
+  
+  while !graph_queue.empty?
+    current = graph_queue.shift
+    if !searched.include?(current)
```

Comment:
I would suggest using a hash with the key being the node and the color being the value.  Then you can do `unless searched[current].nil?`